packages/@sanity/form-builder/src/utils/tests/FormBuilderTester.tsx

```diff
@@ -0,0 +1,86 @@
+//  Schema prop
+// dummy document prop
+
+import {FormFieldPresence} from '@sanity/base/presence'
+import {Marker, Path, Schema, SchemaType} from '@sanity/types'
+import {LayerProvider, studioTheme, ThemeProvider, ToastProvider} from '@sanity/ui'
+import {render} from '@testing-library/react'
+import React from 'react'
+import {FormBuilder} from '../../sanity/legacyPartImplementations/form-builder'
+
+type PatchChannel = {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  subscribe: () => () => {}
+  receivePatches: (patches: any[]) => void
+}
+
+type FormBuilderProps = {
+  value: any | null
+  schema: Schema
+  type: SchemaType
+  markers: Marker[]
+  patchChannel: PatchChannel
+  compareValue: any
+  onFocus: (path: Path) => void
+  readOnly: boolean
+  onChange: (patches: any[]) => void
+  filterField: (field: any) => boolean
+  onBlur: () => void
+  autoFocus: boolean
+  focusPath: Path
+  presence: FormFieldPresence[]
+}
+const patchChannel = FormBuilder.createPatchChannel()
+
+export const DEFAULT_PROPS = {
+  level: 0,
+  markers: [],
+  presence: [],
+  compareValue: undefined,
+  readOnly: undefined,
+  filterField: () => true,
+  patchChannel: patchChannel as any,
+  onBlur: () => undefined,
+  autoFocus: undefined,
+  focusPath: [],
+}
+
+// Use this to test specific inputs rendered in the form-builder
+export function inputTester(document, type, schema, testId?: string) {
```

Comment:
Not super critical, but I'm wondering whether the convenience this offers justifies the added complexity.

Could this be solved by a wrapper around `render` instead?
```js
const {inputContainer} = renderInput('some-test-id', <SomeComponent ... />)
```

Reply:
Yeah I agree... We can for sure improve this, but it's really great for testing all the inputs tho. Maybe we can have a session to look at it, would be nice to establish some patterns we want to use in our testing of the form-builder